aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncClient.java

```diff
@@ -161,6 +163,9 @@ public static AppSyncClient via(@NonNull GraphQLBehavior api) {
             updateInput.put("_version", version);
 
             final Map<String, Object> variables = Collections.singletonMap("input", updateInput);
+            if (predicate != null) {
```

Comment:
I have always wished we had like a non-`null` `QueryPreciates.matchAll()` available. That way, we could avoid having null checks all over the place.